core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceCreator.scala

```diff
@@ -0,0 +1,688 @@
+    * @param span
+    * @return
+    */
+  def createWorkspace(workspaceRequest: WorkspaceRequest, span: Span = null): Future[Workspace] = {
+    validateAttributeNamespace(workspaceRequest.attributes.keys)
+    createWorkspaceInternal(workspaceRequest, span)
+  }
+
+  // TODO: What's up with CloneWorkspace in Orch? Need to make sure this is doing same things as Orch
+  /**
+    * Public method for cloning an existing Workspace.  The creation of the new cloned Workspace will be performed by
+    * `createWorkspaceInternal`, but this method will need to do some work before that to make sure the user has the
+    * necessary permissions on the source Workspace and that any Auth Domain restrictions on the source Workspace are
+    * still in place on the cloned Workspace.  After the cloned Workspace is created, this method will also do things
+    * like copying Entities, Method Configs, and bucket data from the source Workspace into the cloned Workspace.
+    *
+    * @param sourceWorkspaceName
+    * @param destWorkspaceRequest
+    * @return
+    */
+  def cloneWorkspace(sourceWorkspaceName: WorkspaceName, destWorkspaceRequest: WorkspaceRequest): Future[Workspace] = {
+    destWorkspaceRequest.copyFilesWithPrefix.foreach(prefix => validateFileCopyPrefix(prefix))
+    val (libraryAttributeNames, workspaceAttributeNames) = destWorkspaceRequest.attributes.keys.partition(name => name.namespace == AttributeName.libraryNamespace)
+
+    validateAttributeNamespace(workspaceAttributeNames)
+    validateLibraryAttributeNamespaces(libraryAttributeNames)
+
+    for {
+      sourceWorkspace <- getWorkspaceIfHasUserHasAction(sourceWorkspaceName, SamWorkspaceActions.read)
+      destAuthDomains <- validateAuthDomainsForDestWorkspace(sourceWorkspace.workspaceId, destWorkspaceRequest.authorizationDomain.getOrElse(Set.empty))
+      newAttrs = sourceWorkspace.attributes ++ destWorkspaceRequest.attributes
+      cloneWorkspaceRequest = destWorkspaceRequest.copy(authorizationDomain = Option(destAuthDomains), attributes = newAttrs)
+      destWorkspace <- createWorkspaceInternal(cloneWorkspaceRequest)
+      _ <- copyEntitiesAndMethodConfigs(sourceWorkspace, destWorkspace)
+    } yield {
+      //we will fire and forget this. a more involved, but robust, solution involves using the Google Storage Transfer APIs
+      //in most of our use cases, these files should copy quickly enough for there to be no noticeable delay to the user
+      //we also don't want to block returning a response on this call because it's already a slow endpoint
+      copyBucketContents(sourceWorkspace, destWorkspace, cloneWorkspaceRequest.copyFilesWithPrefix)
+
+      destWorkspace
+    }
+  }
+
+  // Temporary method for easily switching between the parallel/sequential versions of createWorkspaceInternal while
+  // this is still in development/testing
+  private def createWorkspaceInternal(workspaceRequest: WorkspaceRequest, span: Span = null): Future[Workspace] = {
```

Comment:
Excellent question.  It is because of when `validateAttributeNamespace` is getting called in create and clone.  

If you look at `cloneWorkspace`, you can see that it calls `validateAttributeNamespace` and `validateLibraryAttributeNamespaces` prior to doing some other "preliminary" stuff before calling `createWorkspaceInternal`.  I did it this way because I didn't dig too deep into what the `validate*AttributeNamespace`actually does and I was generally trying to preserve the ordering of operations from the old code and new.  

This is one of those things where if you can help me confirm that a little reordering of validations and auth checks is OK here, then we can make this change, so please let me know what you think.